jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java

```diff
@@ -70,7 +70,10 @@
             stringToAnnoMap.put(FacesBehavior.class.getName(), FacesBehavior.class);
             stringToAnnoMap.put(FacesBehaviorRenderer.class.getName(), FacesBehaviorRenderer.class);
 
-            // Put JSF 2.2 annotations below this line if any new ones are to be scanned.  So far none.
+            // Put JSF 2.2+ annotations below this line if any new ones are to be scanned.
+            // Load the class to avoid a NoClassDefFoundError if it is not present in the impl
+            ClassLoader loader = AnnotationMap.class.getClassLoader();
+            addAnnotationIfPresent(loader, "javax.faces.view.facelets.FaceletsResourceResolver");
```

Comment:
Could use `FaceletsResourceResolver.class.getName()` here.

Reply:
`FaceletsResourceResolver` is not present in JSF 2.1 and 2.0. If it is imported a `NoClassDefFoundError` is thrown and the Jandex provider is not loaded (default JSF one is used, `DelegatingAnnotationProvider` in 2.1 for example). I discovered this when I did some tests with the multi-jsf. So I decided to do a `loadClass` and don't import the class at all, to make it work with old 2.0 and 2.1 impls. WDYT?

Reply:
Ah, ok, that's fine then.

Reply:
I tested with ee9 too because of the name, and it's translated OK. :smiley:

Reply:
Great, was just thinking about that. Thanks very much, Ricardo!

Reply:
:+